Sources/Sentry/SentryBinaryImageCache.m

```diff
@@ -0,0 +1,121 @@
+#import "SentryBinaryImageCache.h"
+#import "SentryCrashBinaryImageCache.h"
+
+static void binaryImageWasAdded(const SentryCrashBinaryImage *image);
+
+static void binaryImageWasRemoved(const SentryCrashBinaryImage *image);
+
+@implementation BinaryImageInfo
+@end
+
+@interface
+SentryBinaryImageCache ()
+@property (nonatomic, strong) NSMutableArray<BinaryImageInfo *> *cache;
+- (void)binaryImageAdded:(const SentryCrashBinaryImage *)image;
+- (void)binaryImageRemoved:(const SentryCrashBinaryImage *)image;
+@end
+
+@implementation SentryBinaryImageCache
+
++ (SentryBinaryImageCache *)shared
+{
+    static SentryBinaryImageCache *instance = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{ instance = [[self alloc] init]; });
+    return instance;
+}
+
+- (void)start
+{
+    _cache = [NSMutableArray array];
+    sentrycrashbic_registerAddedCallback(&binaryImageWasAdded);
+    sentrycrashbic_registerRemovedCallback(&binaryImageWasRemoved);
+}
+
+- (void)stop
+{
+    sentrycrashbic_registerAddedCallback(NULL);
+    sentrycrashbic_registerRemovedCallback(NULL);
+    _cache = nil;
+}
+
+- (void)binaryImageAdded:(const SentryCrashBinaryImage *)image
```

Comment:
Excellent, this is `O(log n)`. Calling sort after adding an image would be `O(n log n)`, so this is way faster 👍